vsphere/datadog_checks/vsphere/data/conf.yaml.example

```diff
@@ -50,6 +50,24 @@ instances:
     #
     # collect_events: <COLLECT_EVENTS>
 
+    ## @param collect_metric_instance_values - boolean - optional - default: false
+    ## Whether or not to collect metric instance values.
+    ## Metric instance values are send as tag like `metric_instance:my-value`
+    ## An identifier that is derived from configuration names for the device associated with the metric. It identifies
+    ## the instance of the metric with its source. This property may be empty.
+    ##   - For memory and aggregated statistics, this property is empty.
+    ##   - For host and virtual machine devices, this property contains the name of the device, such as the name of the
+    ##     host-bus adapter or the name of the virtual Ethernet adapter. For example, “mpx.vmhba33:C0:T0:L0” or “vmnic0”
+    ##   - For a CPU, this property identifies the numeric position within the CPU core, such as 0, 1, 2, 3.
+    ##   - For a virtual disk, this property identifies the file type:
+    ##       - DISKFILE, for virtual machine base-disk files
+    ##       - SWAPFILE, for virtual machine swap files
+    ##       - DELTAFILE, for virtual machine snapshot overhead files
+    ##       - OTHERFILE, for all other files of a virtual machine
+    ## More info about metric instance: https://code.vmware.com/apis/704/vsphere/vim.PerformanceManager.MetricId.html
+    #
+    # collect_metric_instance_values: false
```

Comment:
Collecting per-instance metric can be really expensive.
For example, if an environment has more than 10k running VM (which is something we need to support), you'll get 10000 times more values to collect (for metrics tagged by VM). Even CPU cores, it is common to have 200 cores on the same ESX host, that means 200 times more values to collect.

I think we should provide more granularity and ask for the metrics you want to collect per instance. (or maybe a mapping of 'resource_type' -> [list of metrics to collect] like we do in `metrics_filters`)

Reply:
Additional note:
When requesting `instance:*`, that means that we will collect and submit both the per-instance values as long as the aggregated value with the `instance:none` tag.
If we ask customers to specifically list the metrics they want to have more granularity on, can we simply ignore the aggregated metric to rely on datadog aggregation?

Reply:
Just saw that part in the PR description:

> - simpler to implement, read, maintain
> - it's consistent with the vSphere documentation: https://code.vmware.com/apis/704/vsphere/vim.PerformanceManager.MetricId.html
> - using different tag keys might lead to confusions and inconsistencies since we need to arbitrarily define them and also maintain them.

1. Agree that makes the code simpler
2. It's consistent because this is an actual limitation on vSphere metrics storage side. From the user experience side, it is much easier to read a meaningful tag key that `metric_instance`
3. I actually think that using different tag keys will help users a lot to avoid confusions, the name of the tag will be enough to understand what it is. Yes we need to arbitrarily define them and maintain them (if they change in the future), but we can always default back to `instance` or `metric_instance` when we don't have an existing mapping.

Reply:
Yes, that's indeed the case. Here is the result:

```
instance='*'
=> Aggregate total of disk used and individual disk used by VM

vsphere.disk.used.latest = 8328192.0, metric_instance:
vsphere.disk.used.latest = 1244593.0, metric_instance:764
vsphere.disk.used.latest = 1244593.0, metric_instance:765
vsphere.disk.used.latest = 1244593.0, metric_instance:766
vsphere.disk.used.latest = 1244593.0, metric_instance:767
=> 1244593 * 4 = 4 978 372
=> 764, 765, 766, 767 are VMs IDs
```

Reply:
> If we ask customers to specifically list the metrics they want to have more granularity on, can we simply ignore the aggregated metric to rely on datadog aggregation?

As you can see above, the aggregated value is not the sum of the other metrics. So we can't count on the backend computation to get the same aggregate.